server/src/main/java/org/cloudfoundry/identity/uaa/provider/ldap/extension/NestedLdapAuthoritiesPopulator.java

```diff
@@ -145,7 +145,7 @@ protected void performNestedSearch(String userDn, String username, Set<GrantedAu
                     role = role.toUpperCase();
                 }
                 role = getRolePrefix() + role;
-                circular = circular | (!authorities.add(new LdapAuthority(role,dn,record)));
+                circular = circular || (!authorities.add(new LdapAuthority(role, dn, record)));
```

Comment:
It does appear that changing `|` to `||` affects the outcome of the executing this line.  In previous code, new `LdapAuthroity` is always added to `authorities`.  In the changed code, new `LdapAuthroity` is not added if `circular` is `true`.

Reply:
hi @hsinn0 i rotated, but it is not wanted to get this merged.

This code is not covered with junit or integration tests yet, so I wont touch it.
I wanted highlight the sonar findings and that sonars marks it as dangerous to use bit-wise OR . 

There is another finding where we working on a fix because there is a test available

Reply:
My understanding of the intention of the existing code is that it wants to always execute `authorities.add(new LdapAuthority(role,dn,record))` and combine the resulting boolean value with `circular`.  So it is proper use of bit-wise OR in my opinion.  Of course, it can be written as:
`circular |= !authorities.add(new LdapAuthority(role,dn,record));  // maybe the intention is clearer this way`
or
`circular = !authorities.add(new LdapAuthority(role,dn,record)) || circular;  // to accomplish the same with ||`